### DISCUSSION TITLE
How to use Using.Manager with Scala 3 given

### DISCUSSION THREAD

**Discussion by AugustNagro** [scala/scala3#18318] [General Discussion]
In Scala 2, it's common to write

```scala
import scala.util.Using

Using.Manager { implicit use =>
  runProgram()
}

def runProgram()(implicit use: Using.Manager): Unit =
  println("hello world")
```

How can we convert this to Scala 3?

**Comment by dwijnand** [comment]
Can you provide a complete piece of code that compiles in Scala 2?

**Reply by AugustNagro** [reply]
Sorry, edited :)

**Comment by bishabosha** [comment]
```scala
inline def managed[T](inline op: Using.Manager ?=> T): Try[T] =
  Using.Manager(op(using _))

managed:
  runProgram()
```

**Reply by AugustNagro** [reply]
That's great, although it would be ideal if users could write something like

```
Using.Manager { given use =>
  runProgram()
}
```

Directly. Or maybe `def managed` could be added to the StdLib eventually.

**Reply by som-snytt** [reply]
`Using.Manager { case given Using.Manager => runProgram() }`

**Reply by AugustNagro** [reply]
@som-snytt very nice
